common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -172,8 +172,12 @@ def _get_modulestore_for_courseid(self, course_id=None):
             if mapping is not None:
                 return mapping
             else:
+                if isinstance(course_id, LibraryLocator):
```

Comment:
It seems a shame to have this logic hidden inside this method. I would move this logic into ``has_course`` and possibly rename it since it isn't necessarily only a course that the store has. Otherwise this sort of tricky logic will be required in other places too, I would imagine.

If it is the wrong thing for ``has_course`` to return true in this case, then I'd argue that it should throw an exception if the locator it receives isn't for a course (if it doesn't already).